Mathlib/LinearAlgebra/TensorProduct.lean

```diff
@@ -206,20 +233,17 @@ theorem smul_tmul [DistribMulAction R' N] [CompatibleSMul R R' M N] (r : R') (m
   CompatibleSMul.smul_tmul _ _ _
 #align tensor_product.smul_tmul TensorProduct.smul_tmul
 
-attribute [local instance] addMonoid
-/-- Auxiliary function to defining scalar multiplication on tensor product. -/
-def SMul.aux {R' : Type*} [SMul R' M] (r : R') : FreeAddMonoid (M × N) →+ M ⊗[R] N :=
-  FreeAddMonoid.lift fun p : M × N => (r • p.1) ⊗ₜ p.2
-#align tensor_product.smul.aux TensorProduct.SMul.aux
-attribute [-instance] addMonoid
+/-- This has the wrong value for the `nsmul` field, but we need it in order to create the right
+value! -/
+private def addCommMonoidWithBadSMul : AddCommMonoid (M ⊗[R] N) where
```

Comment:
Note that the previous `def addMonoid` could have been called `addMonoidWithBadSmul`, so this is no worse than before.